### PR TITLE
chore: slsa 3 provenance generation

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

experimental try out with slsa 3 provenance generation for ko project

cc: @imjasonh @ianlewis

> https://github.com/developer-guy/ko/releases/tag/v0.0.0

### CODE REVIEW

**Comment by imjasonh** [comment]
@laurentsimon possibly superceding https://github.com/google/ko/pull/730 ?

**Comment by laurentsimon** [comment]
We're working on a solution that will help you keep GoReleaser and attach provenance to it, see https://github.com/slsa-framework/slsa-github-generator/tree/main/internal/builders/generic#provenance-with-goreleaser

TL;DR: it will let you build yourself, and the generator will attest to the origin of the repo ("provenance" requirement in SLSA https://slsa.dev/spec/v0.1/requirements). This won't satisfy the "build" requirements of SLSA, but it's still a good improvement.

We'll release this provenance generator this month. Later, when our Go builder has feature parity (or close to) with GoReleaser, we can discuss Go builder :)

Wdut?

**Comment by imjasonh** [comment]
Sounds good to me. I'm glad so many folks are looking into improving the SLSAbility of these release workflows, and I'm happy to just ride that wave 🏄

**Comment by laurentsimon** [comment]
I updated my original https://github.com/google/ko/pull/730 and just realized your PR does the same thing :) We released the generic container today, and it supports uploading the provenance using `upload-assets: true`.

Let me know if I should drop my PR and let this one go thru, or the other way around.
I'm excited either way :)!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -16,8 +18,45 @@ jobs:
...
+
+          checksum_file=$(echo "$ARTIFACTS" | jq -r '.[] | select (.type=="Checksum") | .path')
+          echo "::set-output name=hashes::$(cat $checksum_file | base64 -w0)"
+  provenance:
+    needs: [goreleaser]
+    permissions:
+      actions: read
+      id-token: write
+      contents: read
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.1.1
```
Comment (laurentsimon) [inline .github/workflows/release.yml:42]: v1.2.0 is the release to use to support the generic generator.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -16,8 +18,45 @@ jobs:
...
+    with:
+      base64-subjects: "${{ needs.goreleaser.outputs.hashes }}"
+  
+  # This step creates a GitHub release with our provenance.
+  release-provenance:
```
Comment (laurentsimon) [inline .github/workflows/release.yml:47]: no longer necessary, you can use:
```
with:
  base64-subjects: "${{ needs.goreleaser.outputs.hashes }}"
  upload-assets: true
```

**Comment by imjasonh** [comment]
#730 was merged, closing this

Please feel free to reopen this or open a new issue if there's anything else we should do on top of #730

**Pull request closed without merging**